ragna/core/_components.py

```diff
@@ -76,6 +89,29 @@ def _protocol_model(cls) -> Type[pydantic.BaseModel]:
         return merge_models(cls.display_name(), *cls._protocol_models().values())
 
 
+# Just for demo purposes. We need to move the actual class here.
+# See https://github.com/Quansight/ragna/pull/354#discussion_r1526235318
+class Embedding:
```

Comment:
Let's make this a dataclass if we have no logic on it. We should do the same for `Source` below. We don't need a Pydantic model there.